include/mbedtls/x509_crt.h

```diff
@@ -1108,7 +1108,7 @@ int mbedtls_x509write_crt_set_extension( mbedtls_x509write_cert *ctx,
 int mbedtls_x509write_crt_set_basic_constraints( mbedtls_x509write_cert *ctx,
                                          int is_ca, int max_pathlen );
 
-#if defined(MBEDTLS_SHA1_C)
+#if defined(MBEDTLS_SHA1_C) || ( defined(MBEDTLS_PSA_CRYPTO_C) && defined(PSA_WANT_ALG_SHA_1) )
```

Comment:
This is exactly `MBEDTLS_HAS_ALG_SHA_1_VIA_LOWLEVEL_OR_PSA`

Reply:
Yes, but we can't use macros from the internal header `library/legacy_or_psa.h` in a public header.